R/UploadControls.R

```diff
@@ -605,23 +608,23 @@ join_blood_spot_collections <- function(df.payload, con) {
 #'
 #' @param df.payload A dataframe containing payload data.
 #' @param con A database connection object.
+#' @param count_col Column name in df.payload for count.
+#' @param dbs_control_sheet_id Column name in df.payload for DBS control sheet ID.
 #' @return Result from the dbAppendTable indicating if the collections were added successfully.
-add_new_blood_spot_collections <- function(df.payload, con, count_col) {
-
+add_new_blood_spot_collections <- function(df.payload, con, count_col, dbs_control_sheet_id) {
   res <- dbAppendTable(con, "blood_spot_collection", df.payload %>%
     filter(is.na(blood_spot_collection_id)) %>%
-    group_by(malaria_blood_control_id) %>%
+    group_by(malaria_blood_control_id, dbs_control_sheet_id) %>%
     dplyr::mutate(
       count = as.integer(!!sym(count_col)),
       total = ifelse(is.na(total), 0, as.integer(total)),
       count = sum(count) + total
     ) %>%
     select(-c(total)) %>%
     ungroup() %>%
-    select(malaria_blood_control_id, count) %>%
+    select(malaria_blood_control_id, dbs_control_sheet_id, count) %>%
```

Comment:
Also this may not behave like you think it should, dplyr::select uses non-standard evaluation so it would select the column `"dbs_control_sheet_id"` I think, not a column that matches the string contained inside the variable `dbs_control_sheet_id` defined above as input to the function